InstantPrintTool.py

```diff
@@ -239,6 +243,24 @@ def __export(self):
         if not success:
             QMessageBox.warning(self.iface.mainWindow(), self.tr("Print Failed"), self.tr("Failed to print the composition."))
 
+    def __print(self):
+        self.printer = QPrinter()
+        self.composerView.composition().beginPrint(self.printer)
+        painter = QPainter()
+
+        printdialog = QPrintDialog(self.printer)
+        if printdialog.exec_() != QDialog.Accepted:
+            return
+
+        ready = painter.begin(self.printer)
```

Comment:
You should be able to just use `success = self.composerView.composition().print(self.printer)` instead of lines 255-262